mono/metadata/icall.c

```diff
@@ -1443,30 +1443,24 @@ ves_icall_System_Type_internal_from_name (MonoStringHandle name,
 {
 	error_init (error);
 	MonoTypeNameParse info;
-	gboolean parsedOk;
+	gboolean free_info = FALSE;
```

Comment:
I'm wondering if we should adopt the same idiom of MonoError with MonoTypeNameParse.
IE, we always init it to a state that signals no free'ing is needed.

That would simplify a lot of the usage within the runtime.

Reply:
You know what's funny? we have `MONO_API` functions that consume a `MonoTypeNameParse*` but none that create one, and the underlying struct isn't exposed. So there's no way to actually use, for example, `mono_reflection_parse_type` without access to `metadata-internals.h`.

So yea, I can fixup all the callers on `master` (This PR is for `2017-08`).